### PR TITLE
Optional subsections

### LINKED ISSUE #20: Optional subsections does not work.
Optional subsections does not work for .json files
```fsharp
type SubSection = {
    valueA : string
}

type Settings = {
   subSection : SubSection option
}
```
Reading a json with the subsection in place gives a null value.

### PR DESCRIPTION
**Pull request by queil**
## Proposed Changes

This PR potentially fixes #20 

## Types of changes

What types of changes does your code introduce to FsConfig?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Build and tests pass locally
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/FsConfig.Tests/AppConfig.fs [hunk 1]
```diff
@@ -67,6 +67,35 @@ module ``App Config tests`` =
...
+                    optionalSubsectionNone = None
+                    optionalSubsectionSome = Some({ apiKey = "abc-def"; value = 1 })
+
+                }
+
+                |> Ok
+
+            test <@ appConfig.Get<ConfigWithOptionalSubsection>() = expected @>
+
+        // [<Test>] //TODO
```
Comment (tamizhvendan) [inline tests/FsConfig.Tests/AppConfig.fs]: @queil is it something that you planned to add later?

**Review by tamizhvendan — APPROVED** [review]

File: tests/FsConfig.Tests/AppConfig.fs (hunk 1 above)
Comment (queil) [reply tests/FsConfig.Tests/AppConfig.fs]: @tamizhvendan yes, support for lists of records. I'll remove it from here as the implementation is not as straightforward as I thought.

Comment (queil) [reply tests/FsConfig.Tests/AppConfig.fs]: I've just removed the work in progress bits.

Comment (tamizhvendan) [reply tests/FsConfig.Tests/AppConfig.fs]: Yeah, It's a bit tricky. Thanks for the contribution :)

**Comment by realvictorprm** [comment]
Now the only thing which we would need is a new release that contains this useful change :smile:

**Comment by queil** [comment]
@realvictorprm Released now.

**Comment by realvictorprm** [comment]
> @realvictorprm Released now.

Awesome! Thank you so much! :partying_face:

**Pull request merged**